src/EditorFeatures/CSharpTest/DocumentationComments/CodeFixes/AddDocCommentParamNodeCodeFixProviderTests.cs

```diff
@@ -0,0 +1,677 @@
+    public class AddDocCommentParamNodeCodeFixProviderTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override Tuple<DiagnosticAnalyzer, CodeFixProvider> CreateDiagnosticProviderAndFixer(Workspace workspace)
+        {
+            return new Tuple<DiagnosticAnalyzer, CodeFixProvider>(null, new CSharpAddDocCommentParamNodeCodeFixProvider());
+        }
+
+        private async Task TestAsync(string initial, string expected)
+        {
+            var parseOptions = Options.Regular.WithDocumentationMode(DocumentationMode.Diagnose);
+            await TestAsync(initial, expected, parseOptions: parseOptions, compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddDocCommentParamNode)]
+        public async Task AddsParamTag_NoNodesBefore()
+        {
+            var initial = 
+@"class Program
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name=""j""></param>
+    public void Fizz(int [|i|], int j, int k) {}
+}
+";
+
+            var expected =
+@"class Program
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name=""i""></param>
+    /// <param name=""j""></param>
+    public void Fizz(int i, int j, int k) {}
+}
+";
+            await TestAsync(initial, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddDocCommentParamNode)]
+        public async Task AddsParamTag_NoNodesAfter()
+        {
+            var initial =
+@"class Program
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name=""j""></param>
+    public void Fizz(int i, int j, int [|k|]) {}
+}
+";
+
+            var expected =
+@"class Program
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name=""j""></param>
+    /// <param name=""k""></param>
+    public void Fizz(int i, int j, int k) {}
+}
+";
+            await TestAsync(initial, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddDocCommentParamNode)]
+        public async Task AddsParamTag_NodesBeforeAndAfter()
+        {
+            var initial =
+@"class Program
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name=""i""></param>
+    /// <param name=""k""></param>
+    public void Fizz(int i, int [|j|], int k) {}
+}
+";
+
+            var expected =
+@"class Program
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name=""i""></param>
+    /// <param name=""j""></param>
+    /// <param name=""k""></param>
+    public void Fizz(int i, int j, int k) {}
+}
+";
+            await TestAsync(initial, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddDocCommentParamNode)]
+        public async Task AddsParamTag_NestedInSummaryTag()
+        {
+            var initial =
+@"class Program
+{
+    /// <summary>
+    /// <param name=""j""></param>
+    /// </summary>
+    public void Fizz(int i, int j, int [|k|]) {}
```

Comment:
An alternate approach is to use the warning to trigger, but still have the fix compute all the changes necessary.  There's nothing wrong with that approach.  The real questions is whether we feel that people will want to add each node one at a time, or will they want to do it in one fell swoop.  I'm leaning toward the latter.  But i don't feel strongly.